src/main/java/multithreading/practice/RunnableExample.java

```diff
@@ -0,0 +1,19 @@
+package multithreading.practice;
+
+public class RunnableExample implements Runnable {
+    private final Counter counter;
+
+    public RunnableExample(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        System.out.println("Runnable example started working...");
+        for (int i = 0; i < 100; i++) {
```

Comment:
for the thread run you should increment counter field, not set I all the time new value

Reply:
refactored